engine/vcs/bitbucketcloud/oauth_consumer.go

```diff
@@ -0,0 +1,132 @@
+package bitbucketcloud
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/ovh/cds/sdk"
+	"github.com/ovh/cds/sdk/log"
+)
+
+//AuthorizeRedirect returns the request token, the Authorize Bitbucket cloud
+func (consumer *bitbucketcloudConsumer) AuthorizeRedirect(ctx context.Context) (string, string, error) {
+	requestToken, err := sdk.GenerateHash()
+	if err != nil {
+		return "", "", err
+	}
+
+	val := url.Values{}
+	val.Add("client_id", consumer.ClientID)
+	val.Add("response_type", "code")
+
+	authorizeURL := fmt.Sprintf("https://bitbucket.org/site/oauth2/authorize?%s", val.Encode())
```

Comment:
Can you create const for all bitbucket urls ?

Reply:
Not for this one because it's only used one time but done for acess_token url :)